### PR TITLE
nodeattr -l now supports lists

### PR DESCRIPTION
**Pull request by BenCasses**

### CODE REVIEW

File: src/nodeattr/nodeattr.c [hunk 1]
```diff
@@ -1000,6 +971,70 @@ _hash_attrval(hash_t hattr, char *node, char *attr, char *val)
...
+    hash_t hattrval = NULL;
+    hostlist_t nodelist;
+
+    if ((len = genders_attrlist_create(gp, &attrs)) < 0)
+        _gend_error_exit(gp, "genders_attrlist_create");
+    if ((vlen = genders_vallist_create(gp, &vals)) < 0)
+        _gend_error_exit(gp, "genders_vallist_create");
+    if ((numattrs = genders_getnumattrs(gp)) < 0)
+        _gend_error_exit(gp, "genders_getnumattrs");
+    if (!(hattrval = hash_create((numattrs + 1)*4,
```
Comment (chu11) [inline src/nodeattr/nodeattr.c]: comment to explain the `(numattrs +1)*4` would be good.  I think it's from elsewhere in nodeattr.

File: src/nodeattr/nodeattr.c [hunk 2]
```diff
@@ -1000,6 +971,70 @@ _hash_attrval(hash_t hattr, char *node, char *attr, char *val)
...
+    /* all nodes */
+    } else {
+        if ((count = genders_getattr_all(gp, attrs, len)) < 0)
+            _gend_error_exit(gp, "genders_getattr_all");
+        for (i = 0; i < count; i++)
+            if (node && strlen(vals[i]) > 0)
+                printf("%s=%s\n", attrs[i], vals[i]);
+            else
+                printf("%s\n", attrs[i]);
+    }
```
Comment (chu11) [inline src/nodeattr/nodeattr.c]: logic simplification, if you're in the "all nodes" case, there should be no vals to ever print out?

File: src/nodeattr/nodeattr.c (hunk 1 above)
Comment (BenCasses) [reply src/nodeattr/nodeattr.c]: yes, repeated from elsewhere.  I copied the comments now too

File: src/nodeattr/nodeattr.c (hunk 2 above)
Comment (BenCasses) [reply src/nodeattr/nodeattr.c]: yes, thanks!

**Pull request merged**